Importer

jncchds opened the pull request.

jncchds committed:
Importer

ServAlex commented on DebtBot/Services/ExcelService.cs:
```diff
@@ -0,0 +1,110 @@
+            {
+                try
+                {
+                    date = Parse(worksheet.Cells[row, 1].Value.ToString()!);
```
maybe call it a ParseDate, not clear that we're specifically parse date when reading code

jncchds committed:
Negative amount fix in notification
EnsureBillParticipant is now sync
Dates are parsed in UTC for Excel
Parts are divided by 10000 when they exceed 1mln
Telegram logger has try-catch for message sending

ServAlex commented on DebtBot/Services/BillService.cs:
```diff
@@ -17,375 +17,394 @@
-
-    public BillLineModel? GetLine(Guid id)
-    {
+            .ProjectTo<BillModel>(_mapper.ConfigurationProvider)
```
test this change, seems like we're going to lose spendings in bill details

ServAlex commented on DebtBot/Services/BillService.cs:
```diff
@@ -17,375 +17,394 @@
+        }
+
+        //_publishEndpoint.Publish(new EnsureBillParticipant(bill.Id, creator.Id));
+        ensureBillParticipant(bill.Id, creator.Id);
```
save changes(or move down)  before this line in case bill did not exist, to have an Id

ServAlex commented on DebtBot/Services/BillService.cs:
```diff
@@ -17,375 +17,394 @@
+        }
+
+        _debtContext.SaveChanges();
+
```
or move 
`ensureBillParticipant(bill.Id, creator.Id);`
here

ServAlex commented on DebtBot/Services/BillService.cs:
```diff
@@ -17,375 +17,394 @@
             .Include(q => q.Payments)
             .ThenInclude(q => q.User)
-            .Where(q => q.CreatorId == userId)
+            .Where(q => q.CreatorId == userId || q.Payments.Any(w => w.UserId == userId) || q.Lines.Any(w => w.Participants.Any(e => e.UserId == userId)))
```
this is the method to get bill Created by User
did you mix it up with GetForUser which already has
```
			.Bills
			.Where(b => b
				.BillParticipants
				.Any(p => p.UserId == userId))
```
?
we might need to rename it to `GetCreatedByUser` to not repeat it

ServAlex reviewed (changes requested):
commented concerning aspects

jncchds committed:
OrderBy for Debts, Bills and Spendings
Removed unneeded includes
Renamed Bills.GetByUser to GetCreatedByUser

jncchds committed:
Renamed Parse to ParseDate
EnsureBillParticipant is back via Queue
Separate transaction for Bill creation
